services/core-web/common/utils/Validate.js

```diff
@@ -167,6 +167,22 @@ export const dateNotBeforeOther = memoize((other) => (value) =>
     : undefined
 );
 
+export const dateNotBeforeStrictOther = memoize((other) => (value) =>
+  value && other && new Date(value) < new Date(other) ? `Date cannot be before ${other}` : undefined
+);
+
+export const timeNotBeforeOther = memoize(
+  (comparableDate, baseDate, baseTime) => (comparableTime) =>
+    comparableDate &&
+    comparableTime &&
+    baseDate &&
+    baseTime &&
+    new Date(baseDate) === new Date(comparableDate) &&
```

Comment:
for this to work as intended the following line 180 and 181 need to be a single clause cause it only maters if the the time is before if the dates match